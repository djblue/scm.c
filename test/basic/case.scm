(write (case 'hello
         [(1) 2]
         [else 'hello]))

(write (case (+ 7 5)
         [(1 2 3) 'small]
         [(10 11 12) 'big]))

(write (case (- 7 5)
         [(1 2 3) 'small]
         [(10 11 12) 'big]))

(write (case '(y x)
         [((a b) (x y)) 'forwards]
         [((b a) (y x)) 'backwards]))

(write (case 'x
         [(x) "ex"]
         [('x) "quoted ex"]))
