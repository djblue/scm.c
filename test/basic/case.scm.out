hello
big
small
backwards
"ex"
